python/common/org/python/java/Function.java

```diff
@@ -381,9 +381,10 @@ public static MatchType parameterMatch(java.lang.Class<?> from_type, java.lang.C
     }
 
     public java.lang.Object[] adjustArguments(java.lang.reflect.Method method, org.python.Object[] args, java.util.Map<java.lang.String, org.python.Object> kwargs) {
-        java.lang.Object[] adjusted = new java.lang.Object[args.length];
+        int n_args = (args == null) ? 0 : args.length;
+        java.lang.Object[] adjusted = new java.lang.Object[n_args];
```

Comment:
Similarly - this is an empty array being created.

Reply:
In fact, I don't think there's no need to call this method *at all* if there aren't any arguments.

Reply:
You're absolutely right! Done